### PR TITLE
Comply with new encryption and versioning requirements for S3 and Kinesis

### PR DESCRIPTION
**Pull request by PettitWesley**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: integ/resources/cfn-kinesis-s3-firehose.yml [hunk 1]
```diff
@@ -42,8 +42,18 @@ Resources:
     Type: AWS::Kinesis::Stream
     Properties:
       ShardCount: 5
+      StreamEncryption:
+        EncryptionType: KMS
+        KeyId: alias/aws/kinesis
```
Comment (DrewZhang13) [inline integ/resources/cfn-kinesis-s3-firehose.yml:47]: Have double checked for the keyId and put the official link there if someone else has similar concern about the keyId value is correct or not: https://docs.aws.amazon.com/kinesis/latest/APIReference/API_StartStreamEncryption.html

File: integ/resources/cfn-kinesis-s3-firehose.yml [hunk 2]
```diff
@@ -42,8 +42,18 @@ Resources:
...
   s3Bucket:
     Type: AWS::S3::Bucket
+    Properties:
+      VersioningConfiguration:
+        Status: Enabled
+      BucketEncryption:
+        ServerSideEncryptionConfiguration: 
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: AES256
```
Comment (DrewZhang13) [inline integ/resources/cfn-kinesis-s3-firehose.yml:56]: And the official doc link for this config: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-serversideencryptionbydefault.html

File: integ/resources/cfn-kinesis-s3-firehose.yml [hunk 3]
```diff
@@ -42,8 +42,18 @@ Resources:
...
     Properties:
       ShardCount: 5
+      StreamEncryption:
+        EncryptionType: KMS
+        KeyId: alias/aws/kinesis
   s3Bucket:
     Type: AWS::S3::Bucket
+    Properties:
+      VersioningConfiguration:
+        Status: Enabled
```
Comment (DrewZhang13) [inline integ/resources/cfn-kinesis-s3-firehose.yml:52]: official doc link for this config: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-versioningconfiguration.html

**Review by DrewZhang13 — APPROVED** [review]
Double checked the new configs in PR looks correct to me.

File: integ/resources/cfn-kinesis-s3-firehose.yml (hunk 1 above)
Comment (PettitWesley) [reply integ/resources/cfn-kinesis-s3-firehose.yml:47]: Thanks

**Pull request merged**